sinks/dashboard/items/plot_dash_item.py

```diff
@@ -53,7 +55,20 @@ def add_parameters(self):
         series_param = SeriesChecklistParameter()
         limit_param = {'name': 'limit', 'type': 'float', 'value': 0}
         offset_param = {'name': 'offset', 'type': 'float', 'value': 0}
-        return [series_param, limit_param, offset_param]
+        show_slope_param = {'name': 'Show Slope of Linear Approx.', 'type': 'bool', 'value': False}
+        num_points_param = {'name': 'Slope: Num. of Points in Approx.', 'type': 'int', 'value': 10, 'limits': (2, None), 'visible': False} #How many points to use to calculate slope
+        return [series_param, limit_param, offset_param, show_slope_param, num_points_param]
+    
+    def _calculate_slope(self, times: list[float], points: list[float], num_points: int) -> float:
+        if len(times) < num_points or len(points) < num_points:
+            return np.nan
+        x = np.array(times[-num_points:])
+        y = np.array(points[-num_points:])
```

Comment:
Nit: Data types for the numpy arrays
```suggestion
        x = np.array(times[-num_points:], dtype=np.float64)
        y = np.array(points[-num_points:], dtype=np.float64)
```

Reply:
Or you can alternatively eliminate them altogether potentially by just going np.polyfit(times[-num_points:], points[-num_points:], deg=1)

Reply:
Watch out for the duplicate i don't just mean replace the y LOL.